packages/payload/src/fields/config/types.ts

```diff
@@ -430,21 +430,37 @@ export type SelectField = FieldBase & {
   type: 'select'
 }
 
-export type RelationshipField = FieldBase & {
-  admin?: Admin & {
-    allowCreate?: boolean
-    components?: {
-      Error?: React.ComponentType<ErrorProps>
-      Label?: React.ComponentType<LabelProps>
-    }
-    isSortable?: boolean
+type PolymorphicRelationship = {
+  admin?: {
+    sortOptions?: { [collectionSlug: string]: string }
   }
-  filterOptions?: FilterOptions
-  hasMany?: boolean
-  maxDepth?: number
-  relationTo: string | string[]
-  type: 'relationship'
-} & (
+  relationTo: string[]
+}
+
+type SingleRelationship = {
+  admin?: {
+    sortOptions?: string
+  }
+  relationTo: string
+}
+
+export type RelationshipTypes = PolymorphicRelationship | SingleRelationship
+
+export type RelationshipField = FieldBase &
+  RelationshipTypes & {
+    admin?: Admin & {
```

Comment:
This looks like it would wipe out any `admin` property from `RelationshipTypes`. Is this correct?

Reply:
No its not, there is an issue here for sure. Trying to come up with a good solution, I would like to keep the type narrowing. But I could remove it if its too much effort.

Reply:
@denolfe I made some changes, could you look them over for me? https://github.com/payloadcms/payload/pull/4301/commits/423d1dcffdf4a964c2c88dabcc6f3d8d3bf5c0f9

Reply:
Looks good.